src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/TierCalculationService.kt

```diff
@@ -67,10 +57,122 @@ class TierCalculationService(
     } else {
       log.info("Found latest tier calculation for $crn")
     }
-
     return calculation
   }
 
+  fun calculateProtectLevel(crn: String): TierLevel<ProtectLevel> {
+    val riskPoints = getRiskPoints(crn)
+    val mappaPoints = getMappaPoints(crn)
+    val complexityPoints = getComplexityPoints(crn)
+
+    val totalPoints = riskPoints + mappaPoints + complexityPoints
+    val tier = when {
+      totalPoints >= 30 -> ProtectLevel.A
+      totalPoints in 20..29 -> ProtectLevel.B
+      totalPoints in 10..19 -> ProtectLevel.C
+      else -> ProtectLevel.D
+    }
+
+    return TierLevel(tier, totalPoints)
+  }
+
+  fun calculateChangeLevel(crn: String): TierLevel<ChangeLevel> {
+    val points = getOasysNeedsPoints(crn).plus(getOgrsPoints(crn))
+    val tier = when {
+      points >= 20 -> ChangeLevel.THREE
+      points in 10..19 -> ChangeLevel.TWO
+      else -> ChangeLevel.ONE
+    }
+
+    return TierLevel(tier, points)
+  }
+
+  private fun getRiskPoints(crn: String): Int {
+    return maxOf(getRsrPoints(crn), getRoshPoints(crn))
+  }
+
+  private fun getRsrPoints(crn: String): Int {
+    return communityApiDataService.getRSR(crn).let {
+      when {
+        it != null && it >= RsrThresholds.TIER_B_RSR.num -> 20
+        it != null && it >= RsrThresholds.TIER_C_RSR.num -> 10
+        else -> 0
+      }
+    }
+  }
+
+  private fun getRoshPoints(crn: String): Int {
+    return communityApiDataService.getRosh(crn).let {
+      when (it) {
+        Rosh.VERY_HIGH -> 30
+        Rosh.HIGH -> 20
+        Rosh.MEDIUM -> 10
+        else -> 0
+      }
+    }
+  }
+
+  private fun getMappaPoints(crn: String): Int {
+    return when (communityApiDataService.getMappa(crn)) {
+      Mappa.M2, Mappa.M3 -> 30
+      Mappa.M1 -> 5
+      else -> 0
+    }
+  }
+
+  private fun getComplexityPoints(crn: String): Int {
```

Comment:
Not sure on the readability here, the other WHEN()s are much better

Reply:
as long as it works and the Test coverage is there for all the branches